burn-core/src/data/dataloader/base.rs

```diff
@@ -1,16 +1,24 @@
 pub use crate::data::dataset::{Dataset, DatasetIterator};
 use core::iter::Iterator;
 
+/// A progress struct that can be used to track the progress of a data loader.
 #[derive(Clone, Debug)]
 pub struct Progress {
+    /// The number of items that have been processed.
     pub items_processed: usize,
+
+    /// The total number of items that need to be processed.
     pub items_total: usize,
 }
 
+/// A data loader iterator that can be used to iterate over a data loader.
 pub trait DataLoaderIterator<O>: Iterator<Item = O> {
+    /// Returns the progress of the data loader.
     fn progress(&self) -> Progress;
 }
 
+/// A data loader that can be used to iterate over a dataset.
 pub trait DataLoader<O> {
+    /// Returns the dataset that the data loader is iterating over.
     fn iter<'a>(&'a self) -> Box<dyn DataLoaderIterator<O> + 'a>;
```

Comment:
Not quite right

`/// Iterate over the data loader using a boxed [iterator](DataLoaderIterator)`